protocol/pubsub/v2/protocol.go

```diff
@@ -233,8 +233,14 @@ var _ protocol.Sender = (*Protocol)(nil)
 var _ protocol.Receiver = (*Protocol)(nil)
 var _ protocol.Closer = (*Protocol)(nil)
 
+type withInitialMessageAttributes struct{}
 type withOrderingKey struct{}
 
+// WithMessageAttributes sets the initial Message Attributes for publishing events.
```

Comment:
There's also no way to overwrite/delete them with `SetAttribute` since that assumes the `ce-` prefix.

Reply:
> There's also no way to overwrite/delete them with `SetAttribute` since that assumes the `ce-` prefix.

I assume you're talking about https://github.com/cloudevents/sdk-go/blob/24f0eb47c891bbc5955ed1dd3bd2f4582f6270af/protocol/pubsub/v2/write_pubsub_message.go#L70 right?  I'm probably missing it, but how would someone add a non-CE attribute in general ?

Reply:
Oops, spot on @duglin ! I thought those were for "arbitrary" attributes (not just CE) sorry 🤦 
So, my question would rather be related to extension attributes here because currently we hardcode-prefix them with `ce-` which is not strictly required as per the spec - making it impossible to provide custom extension attributes as discussed in this PR (hence the approach taken in this PR).

Reply:
Well, I'm a bit confused... my normal state these days :-)  but when I look at the test in this PR it appears to be adding non-CE attributes like `Proxy-Authorization`, which seems like a reasonable thing to do. However, w/o this PR how would someone add that attribute in general? I'm trying to make sure we're consistent between how we allow people to define "initial attributes" and "non-initial attributes" for both CE and non-CE attributes. If we only allow people to set non-CE attributes as "initial" ones, that would feel a bit odd to me.

Reply:
Rewording my question: why would we ever prefix (custom) extension attributes with `ce-`?

Reply:
The CE spec doesn't differentiate between spec-defined and extension attributes w.r.t. how they are serialized:
```
Extension attributes are always serialized according to binding rules like standard attributes.
```
So all CE context attributes are prefixed with `ce-` if that's what the protocol bindings defines.

Reply:
> Well, I'm a bit confused... my normal state these days :-) but when I look at the test in this PR it appears to be adding non-CE attributes like `Proxy-Authorization`, which seems like a reasonable thing to do. However, w/o this PR how would someone add that attribute in general? I'm trying to make sure we're consistent between how we allow people to define "initial attributes" and "non-initial attributes" for both CE and non-CE attributes. If we only allow people to set non-CE attributes as "initial" ones, that would feel a bit odd to me.

@AndreasBergmeier6176 thoughts?